sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -634,9 +639,9 @@ def test_gaussian_mixture_fit():
             assert_allclose(ecov.error_norm(prec_pred[k]), 0, atol=0.15)
 
 
-def test_gaussian_mixture_fit_best_params():
-    rng = np.random.RandomState(0)
-    rand_data = RandomData(rng)
+def test_gaussian_mixture_fit_best_params(global_random_seed):
+    rng = np.random.RandomState(global_random_seed)
+    rand_data = RandomData(rng, n_samples=400, n_features=3, scale=100)
```

Comment:
```suggestion
    rand_data = RandomData(rng)
```
There should be no need to change this.

Reply:
I'm not super familiar with the code that is being tested here, it looks like the test compares performing ten individual fits to the result of doing one fit with ten "random inits".

So naively I'd expect that this test should either always pass or always fail. Independent of which random seed is chosen. Do you have an idea why changing the dataset in the way you did helps? From the difference in the results it looks like they are almost but not quite the same. So maybe increasing the size of the dataset will help with statistical fluctuations that lead to the difference? But that requires a bit of knowledge of the algorithm I think.

Reply:
I agree with you that to actually understand the behavior or why the test passes with a small increase in the dataset, I would have to dig deeper into the actual code and working behind the EM algorithm.

Reply:
@betatim I did check this a bit and it seems the randomness is involved in a combination of the data generation and in computing the centroids in Kmeans. So it is possible some random seeds produce a combination which produces slightly different results overall when we compare the individual 10 fits with the single fit with 10 initializations. Therefore for now I just removed the global_random_seed from these two tests.

Reply:
Thanks for digging into this. If we can't make the call to the random number generators exactly the same, so that they get the same sequence of random numbers then we have to allow from some fluctuations. Like you said.

One way out is to increase the dataset, in the hope of reducing the size of the fluctuations.

I think the reason it works with the current setup (what is in `main`), must be pure luck then, right?

I wonder if there is a way to make the test's assert statement more forgiving, basically "the two numbers are equal to within the statistical precision we have" or something like that.

Reply:
I think we can use assert_allclose along with adjusting atol or rtol. However we can also increase the dataset like I had done originally and that passes all the seeds.

Reply:
The random_state was not used correctly. To be comparable, the rng has to be consumed the same way in both cases. I fixed the test and now it passes

Reply:
Also, the test should not have passed in the first place because it was wrong. Since we compare scores, we need to take the max because for a score bigger is better.